rancher2/structure_cluster_v2_rke_config_machine_pool.go

```diff
@@ -97,6 +98,9 @@ func flattenClusterV2RKEConfigMachinePools(p []provisionv1.RKEMachinePool) []int
 		if in.UnhealthyRange != nil {
 			obj["unhealthy_range"] = *in.UnhealthyRange
 		}
+		if in.HostnameLengthLimit >= capr.MinimumHostnameLengthLimit && in.HostnameLengthLimit <= capr.MaximumHostnameLengthLimit {
```

Comment:
The upper and lower limits are treated as valid values. However, they are not on the rancher side https://github.com/rancher/rancher/blob/release/v2.7/pkg/controllers/capr/machineprovision/args.go#L358-L367 

 ```suggestion
 		if in.HostnameLengthLimit > capr.MinimumHostnameLengthLimit && in.HostnameLengthLimit < capr.MaximumHostnameLengthLimit {
```

Reply:
I think you may want to take another peek at the rancher code, as the logic is purposefully inverted there.

Reply:
@jakefhyde, I read the code again and have the following question: 
If the rancher side allows any value and only uses the value if `MinLimit < user-value < MaxLimit`, why do we need to check the value here? Why do we not just pass the value to Rancher and let Rancher determine?

Reply:
@jiaqiluo @a-blender Updated, added support for the cluster wide component (which I forgot) and minor tweaks based on feedback